common/lib/xmodule/xmodule/x_module.py

```diff
@@ -603,8 +603,9 @@ def editable_metadata_fields(self):
         fields = getattr(self, 'unmixed_class', self.__class__).fields
 
         for field in fields.values():
-
-            if field.scope != Scope.settings or field in self.non_editable_metadata_fields:
+            if field in self.non_editable_metadata_fields:
+                continue
+            if field.scope not in (Scope.settings, Scope.content):
```

Comment:
That test is for Advanced Settings though, which is different from the component editors. It doesn't cover this code.